api/query/cache/poll/poll.go

```diff
@@ -84,20 +87,77 @@ func wait(start time.Time, total time.Duration) {
 	}
 }
 
-// KeepMetadataUpdated implements updates to metadataMapCached via simple polling every
+// MetadataPollingService performs metadata-related services via simple polling every
 // interval duration.
-func KeepMetadataUpdated(client *http.Client, logger shared.Logger, interval time.Duration) {
-	logger.Infof("Metadata update via polling started")
+func MetadataPollingService(ctx context.Context, logger shared.Logger, interval time.Duration) {
```

Comment:
Nit; usually a method name that has some result should be a verb-sentence. This is a noun currently, so its not really clear what happens with the 'MetadataPollingService'.

Perhaps: `StartMetadataPollingService`?